src/app/pages/home/hooks/use-onboarding-steps.ts

```diff
@@ -0,0 +1,49 @@
+import { useCallback, useEffect } from 'react';
+
+import {
+  useAccounts,
+  useCurrentAccountAvailableStxBalance,
+} from '@app/store/accounts/account.hooks';
+import {
+  useHideOnboardingStepsState,
+  useOnboardingStepsStatusState,
+} from '@app/store/onboarding/onboarding.hooks';
+import { useCurrentAccountNames } from '@app/query/bns/bns.hooks';
+import { OnboardingSteps, OnboardingStepStatus } from '@shared/models/onboarding-types';
+
+export function useOnboardingSteps() {
+  const [onboardingStepsStatus, setOnboardingStepsStatus] = useOnboardingStepsStatusState();
+  const [hideOnboardingSteps] = useHideOnboardingStepsState();
+  const availableStxBalance = useCurrentAccountAvailableStxBalance();
+  const names = useCurrentAccountNames();
+  const accounts = useAccounts();
+
+  useEffect(() => {
+    if (availableStxBalance?.isGreaterThan(0)) {
+      setOnboardingStepsStatus({
+        ...onboardingStepsStatus,
+        [OnboardingSteps.AddFunds]: OnboardingStepStatus.Done,
+      });
+    }
+
+    if (!!names.length) {
+      setOnboardingStepsStatus({
+        ...onboardingStepsStatus,
+        [OnboardingSteps.RegisterName]: OnboardingStepStatus.Done,
+      });
+    }
+  }, [availableStxBalance, names, onboardingStepsStatus, setOnboardingStepsStatus]);
+
+  const checkIfAllOnboardingStepsAreDone = useCallback(() => {
+    return Object.values(onboardingStepsStatus).every((val, _, arr) => val === arr[0]);
```

Comment:
Wouldn't be safer to check against the static 'Done' state rather than assuming the first item of arr is Done ?

Reply:
I can do that, yep, seems safer. I did it this way bc the first step is always done atm but maybe that will change at some point. 👍